src/transports/tcp.rs

```diff
@@ -189,6 +192,18 @@ impl Transport for TcpTransport {
                         for event in events.iter() {
                             match event.token() {
                                 NEW_CONNECTION => {
+                                    {
+                                        let read_active_connections = active_connections.read();
+                                        let total_in_connections = read_active_connections
+                                        .connections
+                                        .iter()
+                                        .filter(|(_, connection)| connection.connection_type == PeerConnectionType::IN)
+                                        .count() +  read_active_connections
+                                        .connection_queue.len();
+                                        if total_in_connections >= config.max_in_connections {
+                                            continue;
```

Comment:
maybe todo send message to inform remote peer

Reply:
No if we send a message we accept the connections and so we are sensible to flood like now.